### PR TITLE
Adds LLM performance matrix doc to ESS & serverless

### LINKED ISSUE #5238: [Request] Add a page showing model recommendation matric
### Description

What: We want to make users aware of which GenAI models perform best at specific tasks. For this, we should make a concise matrix available for users on its own independent page within security docs. As use cases for AI features within Elastic Security grow, we can expand the matrix to show more use cases.

Given that 8.14 is delayed, this should be made available alongside the release.

### Background & resources

* Docs link: [link](https://docs.google.com/spreadsheets/d/1V6Qofsf2sjyYR6dgkfcS56lVGQXKEIW9seaMasbSd3A/edit?usp=sharing)
* Point of contact: @dhru42 @jamesspi 

### Which documentation set does this change impact?

ESS and serverless

### ESS release

8.14

### Serverless release

Before 8.14 release

### Feature differences

Identical in serverless and ess

### API docs impact

None

### Prerequisites, privileges, feature flags

None

### PR DESCRIPTION
**Pull request by benironside**
Fixes #5238 by adding a new page with a table describing how different large language models perform on various tasks within Elastic Security.

Preview: [LLM performance matrix](https://docs-elastic-7go27535a-elastic-dev.vercel.app/serverless/security/llm-performance-matrix) (Serverless)
[LLM performance matrix](https://security-docs_bk_5286.docs-preview.app.elstc.co/guide/en/security/master/llm-performance-matrix.html) (ESS)

### CODE REVIEW

**Comment by elasticdocs** [comment]
:rocket: Built _co-preview-docs_ successfully!

* https://docs-elastic-5k29f4bs2-elastic-dev.vercel.app

* Built with commit 1430ae41472b9f426383eff60356fe4455519220

> Issues? Visit #next-docs in Slack

**Comment by joepeeples** [comment]
For some reason, the classic version duplicates the table:

image

But I also see that the preview is still building, so maybe it's already been fixed?

**Review by joepeeples — COMMENTED** [review]
Nice clean table! I kinda think the header row with `Feature` | `Model` isn't necessary and adds a bit of clutter, but I don't know if AsciiDoc and Markdown syntax will allow tables with no header row.

**Review by natasha-moore-elastic — COMMENTED** [review]
A couple of nits, otherwise LGTM!

File: docs/assistant/llm-performance-matrix.asciidoc [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[[llm-performance-matrix]]
+= Large language model performance matrix
+
+This table describes the performance of various large language models (LLMs) for different use-cases in {elastic-sec}, based on our internal testing. To learn more about these use-cases, refer to <<attack-discovery, Attack discovery>> or <<security-assistant, AI Assistant>>.
```
Comment (benironside) [inline docs/assistant/llm-performance-matrix.asciidoc:4]:
```suggestion
This table describes the performance of various large language models (LLMs) for different use cases in {elastic-sec}, based on our internal testing. To learn more about these use cases, refer to <<attack-discovery, Attack discovery>> or <<security-assistant, AI Assistant>>.
```

**Review by natasha-moore-elastic — APPROVED** [review]

**Pull request merged**